### PR TITLE
PoA Example (And update to application.ts)

### LINKED ISSUE #8669: PoA Example App
### Description

Like interoperatability and PoS modules, adding a PoA example app for the ease of testing and usage.

### PR DESCRIPTION
**Pull request by Phanco**
### What was the problem?

This PR resolves #8669

### CODE REVIEW

File: examples/poa-mainchain/src/commands/start.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		if (flags['enable-monitor-plugin']) {
+			app.registerPlugin(new MonitorPlugin(), { loadAsChildProcess: true });
+		}
+		if (flags['enable-report-misbehavior-plugin']) {
+			app.registerPlugin(new ReportMisbehaviorPlugin(), { loadAsChildProcess: true });
+		}
+		if (flags['enable-faucet-plugin']) {
+			app.registerPlugin(new FaucetPlugin(), { loadAsChildProcess: true });
+		}
+		if (flags['enable-dashboard-plugin']) {
```
Comment (ishantiw) [inline examples/poa-mainchain/src/commands/start.ts:122]: also add flag for chain connector plugin similar to other apps

File: framework/src/application.ts [hunk 2]
```diff
@@ -176,21 +193,25 @@ export class Application {
...
+				validatorModule.method,
+				posModule.method,
+			);
+			posModule.addDependencies(
+				randomModule.method,
+				validatorModule.method,
+				tokenModule.method,
+				feeModule.method,
+			);
+		}
```
Comment (ishantiw) [inline framework/src/application.ts:214]: we don't want to use it here, we can simply do it under the example app

File: framework/src/modules/poa/module.ts [hunk 3]
```diff
@@ -331,5 +323,14 @@ export class PoAModule extends BaseModule {
...
 		}
+
+		// Pass the required information to the Validators module.
+		await this._validatorsMethod.setValidatorsParams(
+			methodContext,
+			context,
+			currentRoundSnapshot.threshold,
+			currentRoundSnapshot.threshold,
+			currentRoundSnapshot.validators.map(v => ({ address: v.address, bftWeight: v.weight })),
+		);
```
Comment (ishantiw) [inline framework/src/modules/poa/module.ts:334]: not sure why it was moved here for this PR we should not touch the source code, if its a bug then create an issue and separate PR

File: framework/src/application.ts [hunk 4]
```diff
@@ -203,9 +224,28 @@ export class Application {
...
+					token: tokenModule.method,
+					auth: authModule.method,
+					fee: feeModule.method,
+					poa: poaModule.method,
+					random: randomModule.method,
+					reward: rewardModule.method,
+					interoperability: interoperabilityModule.method,
+				},
+			};
+		}
```
Comment (ishantiw) [inline framework/src/application.ts:248]: we should not touch this file, instead what we did here for PoA, we simply do it under `src/app/app.ts` in `poa-mainchain`

**Review by ishantiw — COMMENTED** [review]
Maybe we should use it as sidechain app instead of mainchain as this is more realistic use case for PoA in Lisk ecosystem

File: framework/src/application.ts (hunk 4 above)
Comment (Phanco) [reply framework/src/application.ts:248]: If that's the case do I need to de-register pos from defaultApplication?

File: framework/src/modules/poa/module.ts (hunk 3 above)
Comment (Phanco) [reply framework/src/modules/poa/module.ts:334]: Created
https://github.com/LiskHQ/lisk-sdk/issues/8670

File: examples/poa-sidechain/package.json [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+			"pre-commit": "lint-staged"
+		}
+	},
+	"dependencies": {
+		"@liskhq/lisk-framework-dashboard-plugin": "^0.2.0-alpha.7",
+		"@liskhq/lisk-framework-faucet-plugin": "^0.2.0-alpha.7",
+		"@liskhq/lisk-framework-forger-plugin": "^0.3.0-alpha.7",
+		"@liskhq/lisk-framework-monitor-plugin": "^0.3.0-alpha.7",
+		"@liskhq/lisk-framework-report-misbehavior-plugin": "^0.3.0-alpha.7",
+		"@liskhq/lisk-framework-chain-connector-plugin": "0.1.0-beta.0",
```
Comment (ishantiw) [inline examples/poa-sidechain/package.json:121]: if you use the same version as pos-mainchain then it will not update the yarn.lock file unnecessarily

**Review by ishantiw — COMMENTED** [review]
small comment around yarn.lock file

**Review by ishantiw — APPROVED** [review]

**Pull request merged**